generative-deep-learning/large-language-models/prompting-mistral7B.ipynb

```diff
@@ -0,0 +1,1594 @@
+    "print(f\"Prompt:\\n{prompt}\")\n",
+    "print(\"Response:\")\n",
+    "for text in llm(prompt, max_new_tokens=250, temperature=0.1, stop=\"</s>\", stream=True):\n",
+    "    print(text, end=\"\", flush=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89fa8272-6572-47da-a365-b23ac32be304",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "<div class=\"alert alert-block alert-info\"><b>The model incorrectly states that AWS announced a new chemical solution called \"AWS Graviton2\" in 2019 (you may get a different answer). One approach to combatting LLM hallucination is to give the model an \"out\" by giving it explicit instructions not to make up information and admit when it doesn't know something. Let's try the same question again by by also adding additional instructions in our prompt as follows: </b></div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "b931cf35-e59e-491c-8772-0cda7f021590",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Prompt:\n",
+      "\n",
+      "<s>[INST] What type of new chemical solution AWS announced at re:Invent this year? Do not make up facts. Say I don't know if you have no information about something or unsure. [/INST]    \n",
+      "\n",
+      "Response:\n",
+      "I apologize, but I do not have any specific information on the new chemical solution that AWS announced at re:Invent this year. Can you please provide me with more context or details so that I can assist you better?"
+     ]
+    }
+   ],
+   "source": [
+    "prompt = prompt_template.format(question=\"What type of new chemical solution AWS announced at re:Invent this year? Do not make up facts. Say I don't know if you have no information about something or unsure.\")\n",
+    "print(f\"Prompt:\\n{prompt}\")\n",
+    "print(\"Response:\")\n",
+    "for text in llm(prompt, max_new_tokens=250, temperature=0.1, stop=\"</s>\", stream=True):\n",
+    "    print(text, end=\"\", flush=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "390f21e2-4b5f-4313-86e5-3885b2f10f1d",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "<div class=\"alert alert-block alert-info\"><b>By giving the model clear instructions and an out, we were able to get the model to admit it didn't have enough context to answer the question rather than making up something.</b></div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a63fb206-6661-4ba6-95c5-6369176ff869",
+   "metadata": {},
+   "source": [
+    "# Chatbot\n",
+    "\n",
+    "Let's make a simple chatbot.  There is no special library to include and no setting to apply to the LLM, all we need is prompt engineering!\n",
+    "\n",
+    "Let's use what we know of prompts with in context learning, to create a simple chatbot. "
```

Comment:
If this sentence means In-context learning (reference: [Language Models are Few-Shot Learners](https://arxiv.org/abs/2106.09685)), I recommend to explain it in the above section.